lua/neo-tree/ui/selector.lua

```diff
@@ -0,0 +1,266 @@
+local vim = vim
+local utils = require("neo-tree.utils")
+local log = require("neo-tree.log")
+
+local M = {}
+
+local sep_tbl = function(sep)
+  if type(sep) == "nil" then
+    return {}
+  elseif type(sep) ~= "table" then
+    return { left = sep, right = sep, override = "active" }
+  end
+  return sep
+end
+
+local get_separators = function(source_index, active_index, force_ignore_left, force_ignore_right)
+  local config = require("neo-tree").config
+  local is_active = source_index == active_index
+  local sep = sep_tbl(config.source_selector.separator)
+  if is_active then
+    sep = vim.tbl_deep_extend("force", sep, sep_tbl(config.source_selector.separator_active))
+  end
+  local show_left = sep.override == "left"
+    or (sep.override == "active" and source_index <= active_index)
+    or sep.override == nil
+  local show_right = sep.override == "right"
+    or (sep.override == "active" and source_index >= active_index)
+    or sep.override == nil
+  return {
+    left = (show_left and not force_ignore_left) and sep.left or "",
+    right = (show_right and not force_ignore_right) and sep.right or "",
+  }
+end
+
+M.get_selector_tab_info = function(source_name, source_index, is_active, separator)
+  local config = require("neo-tree").config
+  local separator_config = utils.resolve_config_option(config, "source_selector", nil)
+  if separator_config == nil then
+    log.warn("Cannot find source_selector config. `create_selector` abort.")
+    return {}
+  end
+  local get_strlen = vim.fn.strdisplaywidth
```

Comment:
the api actually provides an alternative that doesn't require a viml round trip i.e. `api.nvim_strwidth`

Reply:
Thanks!